src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts

```diff
@@ -224,6 +245,13 @@ export class GKEClusterSettingsComponent
         .subscribe(_ => {
           this._getGKEKubernetesVersions();
         });
+      this.control(Controls.Zone)
+        .valueChanges.pipe(takeUntil(this._unsubscribe))
```

Comment:
Try to clear disktypes and machine types when new API call is send.

Check `_clearVmSize` in AKSClusterSettingsComponent as a reference